vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -502,9 +502,9 @@ public void setRequired(boolean required) {
     }
 
     @Override
-    public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
+    public void setRequiredIndicatorVisible(boolean required) {
         super.setRequiredIndicatorVisible(required);
-        this.required = requiredIndicatorVisible;
+        this.required = required;
```

Comment:
Will it be more correct ?
`setRequired(required); `
instead of 
`this.required = required;`

Reply:
I don't want to make that change in this PR since `setRequired` might be overridden in some subclass. For the future, I believe `setRequired` should be removed in favor of `setRequiredIndicatorVisible`